src/packages/shared-utils/packageCheck.ts

```diff
@@ -0,0 +1,71 @@
+import { OsMainSourceItem, PlanType, SEATOOL_STATUS } from "../shared-types";
+import { getLatestRai } from "./rai-helper";
+
+const secondClockStatuses = [
+  SEATOOL_STATUS.PENDING,
+  SEATOOL_STATUS.PENDING_APPROVAL,
+  SEATOOL_STATUS.PENDING_CONCURRENCE,
+];
+
+const checkPlan = (planType: PlanType | null, validPlanTypes: PlanType[]) =>
+  !planType
+    ? false
+    : validPlanTypes.includes(planType.toLowerCase() as PlanType);
+
+const checkStatus = (seatoolStatus: string, authorized: string | string[]) =>
+  typeof authorized === "string"
+    ? seatoolStatus === authorized
+    : authorized.includes(seatoolStatus);
+
+/** A object of booleans and methods handling common conditions
+ * for business logic. */
+export const PackageCheck = ({
+  seatoolStatus,
+  rais,
+  raiWithdrawEnabled,
+  planType,
+}: OsMainSourceItem) => {
+  const latestRai = getLatestRai(rais);
+  const planChecks = {
+    isSpa: checkPlan(planType, [PlanType.MED_SPA, PlanType.CHIP_SPA]),
+    isWaiver: checkPlan(planType, []),
+    /** Keep excess methods to a minimum with `is` **/
+    planTypeIs: (validPlanTypes: PlanType[]) =>
+      checkPlan(planType, validPlanTypes),
+  };
+  const statusChecks = {
+    /** Is in any of our pending statuses, sans Pending-RAI **/
+    isInActivePendingStatus: checkStatus(seatoolStatus, [
+      ...secondClockStatuses,
+      SEATOOL_STATUS.PENDING_OFF_THE_CLOCK,
+    ]),
+    /** Is in a second clock status and RAI has been received **/
+    isInSecondClock:
+      !planChecks.planTypeIs([PlanType.CHIP_SPA]) &&
+      checkStatus(seatoolStatus, secondClockStatuses) &&
+      latestRai?.status === "received",
+    /** Is in any status except Package Withdrawn **/
+    isNotWithdrawn: !checkStatus(seatoolStatus, SEATOOL_STATUS.WITHDRAWN),
+    /** Added for elasticity, but common checks should always bubble up as
+     * object attributes! **/
+    hasStatus: (authorizedStatuses: string | string[]) =>
+      checkStatus(seatoolStatus, authorizedStatuses),
+  };
+  const raiChecks = {
+    /** Latest RAI is requested and status is Pending-RAI **/
+    hasRequestedRai: latestRai?.status === "requested",
+    /** Latest RAI is not null **/
+    hasLatestRai: latestRai !== null,
+    /** Latest RAI has been responded to **/
+    hasRaiResponse: latestRai?.status === "received",
+    /** RAI Withdraw has been enabled **/
+    hasEnabledRaiWithdraw: raiWithdrawEnabled,
+  };
```

Comment:
I do have some concerns on how well this will respond as more conditions are added.  
But, I trust you have the best visibility and technical insight on the business requirements of packages.  
So, LGTM.  Lets give it a try 👍.

Reply:
Yeah, same. I considered lazy-loading them in _this_ PR for that, but since this was already just a clean-up one commit and the testing had passed on the feature branch, I wanted to save that as a next step when needed, but still be set up with some easy partitions based on what I experienced this go-around.

Namely, the partitions seem to make sense between: Package Details (plan type and status checks) and RAI meta/details. However, _both_ are used in the Actions checking which is the primary first use for this.

Thoughts on that future lazy loading. We can split out the objects into their own classes for use in contexts where that's all we need, like checking package details, but when we have larger aggregates, having a lazy-loader built in to only load the check classes we're using within the parent class:

```typescript
let raiChecks = PackageCheck(data) // Loads the main service
raiChecks.hasLatestRai // Invalid, undefined, null, what have you ❌ 
raiChecks = raiChecks.rai() // Loads the RAI checks with data given to PackageCheck
raiChecks.hasLatestRai // Now a defined boolean value ✅ 
```

Also calling up the former question of using boolean properties or methods returning booleans, if we don't gain much convenience via lazy loads, returning all checks as methods may be a performance buff if needed. I'd LOVE to find a pattern that loads an empty check class but then as each check is USED, if it's undefined, the method loads the data based on the condition, and then a property is set so you _don't_ need to run it again. We just have to very opposite uses: one where we use a ton of checks once or twice per-scope, and one where we use 2-3 checks a bunch per-scope.